zerver/lib/emoji.py

```diff
@@ -62,17 +62,29 @@ def translate_emoticons(text: str) -> str:
     return translated
 
 
-def emoji_name_to_emoji_code(realm: Realm, emoji_name: str) -> Tuple[str, str]:
-    # TODO: Just ask callers for realm_id. Also consider returning a
-    #       tiny dataclass instead of a tuple.
-    realm_emoji_dict = get_realm_active_emoji(realm.id)
+@dataclass
+class EmojiData:
+    emoji_code: str
+    reaction_type: str
+
+
+def get_emoji_data(realm_id: int, emoji_name: str) -> EmojiData:
```

Comment:
How about `get_emoji_data_for_name`?

Reply:
I pretty strongly prefer brevity here.  We don't have any similar functions where the distinction would be confusing.  The original name of this function was overly precise, which led to it eventually being wrong.  I'd rather be brief and vague than overly precise and wrong.

The whole emoji system is kind of confusing with the names and codes and ids, but I would rather address this with big scary comments (like we already have in a couple of the views) than with naming.  And, of course, whoever takes the next pass at this can always improve the name based on newer knowledge or requirements.